<launch>

  <arg name="control_performance_analysis_param_path" default="$(find-pkg-share control_performance_analysis)/config/control_performance_analysis.param.yaml" />
  <arg name="input/reference_trajectory" default="/planning/scenario_planning/trajectory" />
  <arg name="input/control_raw" default="/control/trajectory_follower/lateral/control_cmd" />
  <arg name="input/measured_steering" default="/vehicle/status/steering_status" />
  <arg name="input/current_odometry" default="/localization/kinematic_state" />
  <arg name="output/error_stamped" default="/control_performance/performance_vars" />

  <!-- vehicle info -->
  <include file="$(find-pkg-share autoware_launch)/launch/global_params.launch.py" />

  <node pkg="control_performance_analysis" exec="control_performance_analysis" name="control_performance_analysis" output="screen">
    <param from="$(var control_performance_analysis_param_path)" />

    <remap from="~/input/reference_trajectory" to="$(var input/reference_trajectory)" />
    <remap from="~/input/control_raw" to="$(var input/control_raw)" />
    <remap from="~/input/measured_steering" to="$(var input/measured_steering)" />
    <remap from="~/input/current_odometry" to="$(var input/current_odometry)" />
    <remap from="~/output/error_stamped" to="$(var output/error_stamped)" />
  </node>
</launch>
